builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -244,6 +249,25 @@ func resourceCloudStackInstanceUpdate(d *schema.ResourceData, meta interface{})
 
 	name := d.Get("name").(string)
 
+	if d.HasChange("name") {
```

Comment:
The instance has to be stopped/started for a name change to take effect. So could you move this func to into this func:

```go
// Attributes that require reboot to update
if d.HasChange("service_offering") || d.HasChange("keypair") {
  ...
}
```

Somewhere on line 301 would be a good place for this one...